realm/src/main/java/io/realm/Realm.java

```diff
@@ -421,6 +250,7 @@ private static Realm create(RealmConfiguration configuration) {
     }
 
     private static synchronized Realm createAndValidate(RealmConfiguration configuration, boolean validateSchema, boolean autoRefresh) {
+
```

Comment:
I feel that all the `synchronized` locks on `Realm.class` should be on `BaseRealm.class` instead.